cppForSwig/BlockDataMap.cpp

```diff
@@ -44,15 +44,24 @@ void BlockData::deserialize(const uint8_t* data, size_t size,
    {
       //light tx deserialization, just figure out the offset and size of
       //txins and txouts
-      vector<size_t> offsetIns, offsetOuts;
+      vector<size_t> offsetIns, offsetOuts, offsetsWitness;
       auto txSize = BtcUtils::TxCalcLength(
          brr.getCurrPtr(), brr.getSizeRemaining(),
-         &offsetIns, &offsetOuts);
+         &offsetIns, &offsetOuts, &offsetsWitness);
 
       //create BCTX object and fill it up
       shared_ptr<BCTX> tx = make_shared<BCTX>(brr.getCurrPtr(), txSize);
       tx->version_ = READ_UINT32_LE(brr.getCurrPtr());
 
+      // Marker and flag if it is a witness transaction
```

Comment:
generally, witness data is no needed for scanning, there should be no witness code here, besides figuring out the market + flag